spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -187,33 +190,40 @@ protected void doStart() {
 	@Override
 	protected void doStop() {
 		this.topicLock.lock();
-		String[] topics = getTopic();
+		initializeMqttAsyncClientIfRequired();
```

Comment:
That's something wrong in the `stop()` logic.
if there is not client initialized why would one wants to initialize it form stopping?